### PR TITLE
kevm_pyk: add error logs

### PR DESCRIPTION
**Pull request by anvacaru**
unblocks: https://github.com/runtimeverification/foundry/issues/10
fixes: #1322

### CODE REVIEW

File: kevm_pyk/src/kevm_pyk/solc_to_k.py [hunk 1]
```diff
@@ -101,18 +101,27 @@ def _get_method_abi(_mname: str) -> Dict:
...
+        try:
+            _fields_list = [(_f['label'], int(_f['slot'])) for _f in contract_json['storageLayout']['storage']]
+        except KeyError:
+            _LOGGER.error(f'\'storageLocation\' not found on contract {self.name}')
+            _fields_list = []
         _fields = {}
         for _l, _s in _fields_list:
             if _l in _fields:
-                raise ValueError(f'Found duplicate field access key on contract {self.name}: {_l}')
+                _LOGGER.error(f'Found duplicate field access key on contract {self.name}: {_l}')
```
Comment (anvacaru) [inline kevm_pyk/src/kevm_pyk/solc_to_k.py:123]: this should be a warning

File: kevm_pyk/src/kevm_pyk/solc_to_k.py [hunk 2]
```diff
@@ -101,18 +101,27 @@ def _get_method_abi(_mname: str) -> Dict:
...
             mname = msig.split('(')[0]
             mid = int(method_identifiers[msig], 16)
             _methods.append(Contract.Method(mname, mid, _get_method_abi(mname), contract_name, self.sort_method))
         self.methods = tuple(_methods)
-        _fields_list = [(_f['label'], int(_f['slot'])) for _f in contract_json['storageLayout']['storage']]
+        try:
+            _fields_list = [(_f['label'], int(_f['slot'])) for _f in contract_json['storageLayout']['storage']]
+        except KeyError:
+            _LOGGER.error(f'\'storageLocation\' not found on contract {self.name}')
+            _fields_list = []
```
Comment (ehildenb) [inline kevm_pyk/src/kevm_pyk/solc_to_k.py:119]:
```suggestion
        if 'storageLayout' not in contract_json or 'storage' not in contract_json['storageLayout']:
            _LOGGER.warning(f'Did not find proper storage layout information for contract: {contract.name}')
        else:
            _fields_list = [(_f['label'], int(_f['slot'])) for _f in contract_json['storageLayout']['storage']]
```

File: kevm_pyk/src/kevm_pyk/solc_to_k.py [hunk 3]
```diff
@@ -101,18 +101,27 @@ def _get_method_abi(_mname: str) -> Dict:
 
         self.name = contract_name
         self.bytecode = (contract_json['evm']['deployedBytecode']['object'] if not foundry else contract_json['deployedBytecode']['object'])
-        method_identifiers = contract_json['evm']['methodIdentifiers'] if not foundry else contract_json['methodIdentifiers']
+        try:
+            method_identifiers = contract_json['evm']['methodIdentifiers'] if not foundry else contract_json['methodIdentifiers']
+        except KeyError:
+            _LOGGER.error(f'\'methodIdentifiers\' not found on contract {self.name}')
+            method_identifiers = []
```
Comment (ehildenb) [inline kevm_pyk/src/kevm_pyk/solc_to_k.py:108]: Same here, it seems we can do this with an if-else and a `_LOGGER.warning` instead.

File: kevm_pyk/src/kevm_pyk/__main__.py [hunk 4]
```diff
@@ -74,11 +74,15 @@ def main():
...
+                counter: int = 0
                 # Must sort to get consistent output order on different platforms.
                 for json_file in sorted(glob.glob(path_glob)):
                     _LOGGER.info(f'Processing contract file: {json_file}')
                     contract_name = json_file.split('/')[-1]
                     contract_name = contract_name[0:-5] if contract_name.endswith('.json') else contract_name
+                    if(contract_name.upper() in [module.name.split('-')[0] for module in modules]):
+                        counter += 1
+                        contract_name += str(counter)
                     with open(json_file, 'r') as cjson:
```
Comment (ehildenb) [inline kevm_pyk/src/kevm_pyk/__main__.py:86]: Let's remove this change for now.

**Comment by ehildenb** [comment]
`make tests/foundry/foundry.k.check CHECK=cp` to update the expected file.

File: kevm_pyk/src/kevm_pyk/__main__.py (hunk 4 above)
Comment (anvacaru) [reply kevm_pyk/src/kevm_pyk/__main__.py]: Removed.

File: kevm_pyk/src/kevm_pyk/solc_to_k.py (hunk 3 above)
Comment (anvacaru) [reply kevm_pyk/src/kevm_pyk/solc_to_k.py]: Done.

File: kevm_pyk/src/kevm_pyk/solc_to_k.py (hunk 2 above)
Comment (anvacaru) [reply kevm_pyk/src/kevm_pyk/solc_to_k.py]: Done.

File: kevm_pyk/src/kevm_pyk/solc_to_k.py [hunk 5]
```diff
@@ -101,18 +101,27 @@ def _get_method_abi(_mname: str) -> Dict:
...
+        else:
+            _method_identifiers = contract_json['evm']['methodIdentifiers'] if not foundry else contract_json['methodIdentifiers']
         _methods = []
-        for msig in method_identifiers:
+        for msig in _method_identifiers:
             mname = msig.split('(')[0]
-            mid = int(method_identifiers[msig], 16)
+            mid = int(_method_identifiers[msig], 16)
             _methods.append(Contract.Method(mname, mid, _get_method_abi(mname), contract_name, self.sort_method))
         self.methods = tuple(_methods)
```
Comment (ehildenb) [inline kevm_pyk/src/kevm_pyk/solc_to_k.py:114]: We probably want to just move this entire block into the `else ...` statement.

File: kevm_pyk/src/kevm_pyk/solc_to_k.py [hunk 6]
```diff
@@ -101,18 +101,27 @@ def _get_method_abi(_mname: str) -> Dict:
...
+            _fields_list = []
+        else:
+            _fields_list = [(_f['label'], int(_f['slot'])) for _f in contract_json['storageLayout']['storage']]
         _fields = {}
         for _l, _s in _fields_list:
             if _l in _fields:
-                raise ValueError(f'Found duplicate field access key on contract {self.name}: {_l}')
+                _LOGGER.warning(f'Found duplicate field access key on contract {self.name}: {_l}')
+                continue
             _fields[_l] = _s
```
Comment (ehildenb) [inline kevm_pyk/src/kevm_pyk/solc_to_k.py:125]: Same here, the entire block can be moved into else statement.

File: kevm_pyk/src/kevm_pyk/solc_to_k.py [hunk 7]
```diff
@@ -101,20 +101,29 @@ def _get_method_abi(_mname: str) -> Dict:
...
-        self.methods = tuple(_methods)
-        _fields_list = [(_f['label'], int(_f['slot'])) for _f in contract_json['storageLayout']['storage']]
-        _fields = {}
-        for _l, _s in _fields_list:
-            if _l in _fields:
-                raise ValueError(f'Found duplicate field access key on contract {self.name}: {_l}')
-            _fields[_l] = _s
-        self.fields = FrozenDict(_fields)
+        if 'methodIdentifiers' not in contract_json or not(foundry or 'methodIdentifiers' in contract_json['evm']):
+            _LOGGER.warning(f'\'methodIdentifiers\' not found on contract {self.name}')
```
Comment (ehildenb) [inline kevm_pyk/src/kevm_pyk/solc_to_k.py:105]:
```suggestion
            _LOGGER.warning(f'Could not find member \'methodIdentifiers\' while processing contract: {self.name}')
```

File: kevm_pyk/src/kevm_pyk/solc_to_k.py [hunk 8]
```diff
@@ -101,20 +101,29 @@ def _get_method_abi(_mname: str) -> Dict:
...
+        else:
+            _method_identifiers = contract_json['evm']['methodIdentifiers'] if not foundry else contract_json['methodIdentifiers']
+            _methods = []
+            for msig in _method_identifiers:
+                mname = msig.split('(')[0]
+                mid = int(_method_identifiers[msig], 16)
+                _methods.append(Contract.Method(mname, mid, _get_method_abi(mname), contract_name, self.sort_method))
+            self.methods = tuple(_methods)
+        if 'storageLayout' not in contract_json or 'storage' not in contract_json['storageLayout']:
+            _LOGGER.warning(f'\'storageLocation\' not found on contract {self.name}')
```
Comment (ehildenb) [inline kevm_pyk/src/kevm_pyk/solc_to_k.py:116]:
```suggestion
            _LOGGER.warning(f'Could not find member \'storageLayout\' while processing contract: {self.name}')
```

**Review by ehildenb — APPROVED** [review]

**Pull request merged**